packages/theia-integration/src/browser/diagram/glsp-diagram-widget.ts

```diff
@@ -151,7 +151,7 @@ export class GLSPDiagramWidget extends DiagramWidget implements SaveableSource {
     listenToFocusState(shell: ApplicationShell): void {
         this.toDispose.push(shell.onDidChangeActiveWidget(event => {
             const focusedWidget = event.newValue;
-            if (this.hasFocus && !this.isThisWidget(focusedWidget)) {
+            if (this.hasFocus && focusedWidget && !this.isThisWidget(focusedWidget)) {
```

Comment:
Maybe its better to directly include the `undefined` check into the `isThisWidget()` guard function? e.g:

```typescript
protected isThisWidget(widget?: Widget | null): boolean {
        // eslint-disable-next-line no-null/no-null
        if (!widget || widget === null) {
            return false;
        }
        const diagramWidget = getDiagramWidget(widget);
        return diagramWidget !== undefined && diagramWidget.id === this.id;
    }
```

Reply:
Yeah, but that is not the same behavior, is it?
We want to omit checking whether it is this.isthisWidget() altogether if it is null. In your version, if focusedWidget is null, we'd return false and send out the FocusStateChangedAction even though the focusedWidget was null and no other widget.

Reply:
Ah I see. i overlooked the ! operator in front of `this.isThisWidget` :smile:.
In that case, ignore my comment and lets merge this :wink: